src/dialog/unix/SDL_zenitydialog.c

```diff
@@ -36,34 +36,12 @@ typedef struct
     bool allow_many;
     SDL_FileDialogType type;
     /* Zenity only works with X11 handles apparently */
-    Uint64 x11_window_handle;
+    char x11_window_handle[64];
     const char *title;
```

Comment:
These should be `char *` now, since we're saving a locally allocated pointer. This will allow you to remove the casts in the free function as well.

Reply:
I think the args should all be `const char **`, and if we need to track arguments for freeing, they should be tracked separately.

Very offhand:
```c
const char **args = (const char **)SDL_malloc(...);
... count filter args
char **filter_args = (char **)SDL_malloc(...);
... fill in filter args
for (int i = 0; i < num_filter_args; ++i) {
    args[nargs++] = filter_args[i];
}
...
free filter args
free args as a single pointer
```
This is a little more verbose, but it'll avoid mistakes if someone goes to modify this code in the future.

Reply:
It also means that const warnings from the compiler are providing meaningful feedback, which helps prove correctness.

Reply:
FWIW, the strdup'ing of static strings was there to avoid casting away the const-ness of static strings. It also avoids assigning `char *` to `const char *` in the argument list.